packages/drawer/index.tsx

```diff
@@ -47,15 +48,20 @@ export interface DrawerProps extends React.HTMLProps<HTMLElement>{
   tag?: string;
   dismissible?: boolean;
   modal?: boolean;
+  innerRef?: RefCallback<HTMLElement> | React.RefObject<HTMLElement>;
 };
 
 interface DrawerState {
   classList: Set<string>;
 };
 
+const isRefObject = function(ref: DrawerProps['innerRef']): ref is React.RefObject<HTMLElement> {
+  return typeof ref !== 'function';
+};
+
 class Drawer extends React.Component<DrawerProps, DrawerState> {
   previousFocus: HTMLElement | null = null;
-  foundation: MDCDismissibleDrawerFoundation | MDCModalDrawerFoundation;
+  foundation!: MDCDismissibleDrawerFoundation | MDCModalDrawerFoundation;
```

Comment:
I don't think we should do this in drawer mainly because permanent doesn't have a foundation. I changed the `!` --> `?`. This only changed threw a TS error in componentDidUpdate. Can we add a foundation check there? I think that should be all.

Reply:
okay, will do

Reply:
I think I'll need to update the tests to, adding a `!`.